### PR TITLE
Upgrade terraform-provider-aws to v5.27.0

### LINKED ISSUE #3053: Upgrade terraform-provider-aws to v5.27.0
Release details: https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.27.0

### PR DESCRIPTION
**Pull request by guineveresaenger**
This PR was generated via `$ upgrade-provider pulumi/pulumi-aws`.

---

- Upgrading terraform-provider-aws from 5.26.0  to 5.27.0.
	Fixes #3053
- Upgrading pulumi-terraform-bridge from v3.66.1-0.20231127160738-2f132e750bb5 to v3.66.0.
- Upgrading pulumi-terraform-bridge/pf from v0.20.1-0.20231127160738-2f132e750bb5 to v0.20.0.

### CODE REVIEW

File: provider/resources.go [hunk 1]
```diff
@@ -708,10 +708,6 @@ func ProviderFromMeta(metaInfo *tfbridge.MetadataInfo) *tfbridge.ProviderInfo {
 					EnvVars: []string{"AWS_REGION", "AWS_DEFAULT_REGION"},
 				},
 			},
-			"skip_get_ec2_platforms": {
```
Comment (iwahbe) [inline provider/resources.go:711]: I'm assuming that this was removed upstream, and so warned during the build => removed.

File: provider/resources.go [hunk 2]
```diff
@@ -1070,12 +1066,6 @@ func ProviderFromMeta(metaInfo *tfbridge.MetadataInfo) *tfbridge.ProviderInfo {
 						// Explicitly map tag => tags to avoid confusion with tags => tagsCollection below.
 						Name: "tags",
 					},
-					"tags": {
-						// Conflicts with the pluralized `tag` property, which is the more strongly typed option for
-						// providing tags.  We keep this dynamically typed collection of tags as an option as well, but
-						// give it a different name.
-						Name: "tagsCollection",
-					},
```
Comment (iwahbe) [inline provider/resources.go:1078]: Why was this removed? I don't see a corresponding removal in upstream's release notes.

File: provider/resources.go [hunk 3]
```diff
@@ -1127,16 +1117,6 @@ func ProviderFromMeta(metaInfo *tfbridge.MetadataInfo) *tfbridge.ProviderInfo {
...
-			// Budgets
-			"aws_budgets_budget": {
-				Tok: awsResource(budgetsMod, "Budget"),
-				Fields: map[string]*tfbridge.SchemaInfo{
-					"cost_filters": {
-						Name:               "costFilterLegacy",
-						DeprecationMessage: "The now-deprecated original cost filters. Use CostFilters instead.",
-					},
-				},
-			},
```
Comment (iwahbe) [inline provider/resources.go:1139]: Is this a no-op?

**Review by iwahbe — COMMENTED** [review]
Can we remove the bridge and pf changes. That reverts part of https://github.com/pulumi/pulumi-aws/pull/3047.

I don't understand the changes to `resources.go`. Patch changes all look good.

File: provider/resources.go (hunk 3 above)
Comment (guineveresaenger) [reply provider/resources.go:1139]: No. This is an outdated mapping for `cost_filters` which [does not exist upstream](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/budgets_budget) - note that `cost_filter`, singular, does very much exist. I removed the resource entirely to be picked up by auto resourcing.

File: provider/resources.go (hunk 2 above)
Comment (guineveresaenger) [reply provider/resources.go:1078]: I believe this was removed a while back: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group
- there is no `tags` field. We could also remove the (TF)`tag` -> (pulumi)`tags` mapping but I imagine that would break users.

File: provider/resources.go (hunk 1 above)
Comment (guineveresaenger) [reply provider/resources.go:711]: This was removed on v5.0.0: https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.0.0
and no longer appears on https://registry.teraform.io/providers/hashicorp/aws/latest/docs

File: provider/resources.go (hunk 3 above)
Comment (guineveresaenger) [reply provider/resources.go:1139]: update: This field was removed on [upstream v5.0.0](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.0.0)

File: provider/resources.go (hunk 2 above)
Comment (guineveresaenger) [reply provider/resources.go:1078]: update: like the other removals, this was removed for upstream v5. https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.0.0

**Review by iwahbe — APPROVED** [review]
